controllers/providers/kubernetes/nodes.go

```diff
@@ -32,3 +38,45 @@ func (k *KubernetesClientSet) ListClusterNodes() (*corev1.NodeList, error) {
 	}
 	return nodes, nil
 }
+
+// DrainNode cordons and drains a node.
+func (k *KubernetesClientSet) DrainNode(node *corev1.Node, PostDrainDelaySeconds time.Duration, client kubernetes.Interface) error {
+	if client == nil {
+		return fmt.Errorf("K8sClient not set")
+	}
+
+	if node == nil {
+		return fmt.Errorf("node not set")
+	}
+
+	helper := &drain.Helper{
+		Client:              client,
+		Force:               true,
+		GracePeriodSeconds:  -1,
+		IgnoreAllDaemonSets: true,
+		Out:                 os.Stdout,
+		ErrOut:              os.Stdout,
+		DeleteEmptyDirData:  true,
+		Timeout:             time.Duration(120) * time.Second,
+	}
+
+	if err := drain.RunCordonOrUncordon(helper, node, true); err != nil {
+		if apierrors.IsNotFound(err) {
+			return err
+		}
+		return fmt.Errorf("error cordoning node: %v", err)
+	}
+
+	if err := drain.RunNodeDrain(helper, node.Name); err != nil {
+		if apierrors.IsNotFound(err) {
+			return err
+		}
+		return fmt.Errorf("error draining node: %v", err)
+	}
+
+	if PostDrainDelaySeconds > 0 {
```

Comment:
This should be done via requeue and comparison of timestamp - no more sleeping :)

Reply:
We can do this in a separate story where we handle parallelizing this

Reply:
Ah! Yeah. I think we are already doing that as I am setting the lastNodeDrainTime. I shall remove this sleep.